benchmarks/benchmarks/stats.py

```diff
@@ -434,3 +434,16 @@ def time_fni(self, distcase):
             sup.filter(RuntimeWarning, "divide by zero")
             sup.filter(RuntimeWarning, "invalid value encountered")
             stats.NumericalInverseHermite(dist)
+
+
+class DistanceFunctions(Benchmark):
+    def setup(self):
+        rng = np.random.default_rng(12345678)
+        self.u_values, self.u_weights = rng.random(4) * 10, rng.random(4) * 10
+        self.v_values, self.v_weights = rng.random(2) * 10, rng.random(2) * 10
```

Comment:
It looks to me like the array sizes are so small here that you're only measuring function overhead. If I change it to:
```python
        size = 4000
        rng = np.random.default_rng(12345678)
        self.u_values = rng.random(size) * 10
        self.u_weights = rng.random(size) * 10
        self.v_values = rng.random(size // 2) * 10
        self.v_weights = rng.random(size // 2) * 10
```
then I see that for size `4` or `40` the benchmark takes about 40 us, for `400` it takes ~100 us, and for `4000` it takes ~800 us. So you may want to make this a lot larger (maybe ~10,000?). 

You'll also want to check that the size is such that the performance improvements with Pythran you're trying to show actually show up - in gh-14154 it looks like for very small arrays the gain is ~3x, but what about large arrays? Parametrizing the benchmark on size (e.g. one small size, one large one) may make sense.

Reply:
Thanks for reminding me about the size!  I just found the size is really a big problem... When I run the benchmark with `size = 4000`, the master branch is ~1.32ms while my branch is ~3.39ms. Seems in this case, a larger input size would make pythran slower than the original code... I just submitted an issue in Pythran https://github.com/serge-sans-paille/pythran/issues/1793